packages/ui/src/Radio.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, FocusEvent, useRef } from 'react'
+import styles from './Radio.module.scss'
+import classNames from 'classnames'
+import { Help, helpTooltipId } from './Help'
+import { v4 as uuid } from 'uuid'
+import { DataTestProp } from '@hazelcast/helpers'
+import { RadioGroupContext } from './RadioGroupContext'
+
+type RadioCoreProps = {
+  value: string
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  checked?: boolean
+}
+
+export type RadioExtraProps = {
+  label: string
+  helperText?: string
+  disabled?: boolean
+  required?: boolean
+  className?: string
+}
+
+export type RadioProps = RadioCoreProps & RadioExtraProps & DataTestProp
+
+/**
+ * ### Purpose
+ * Forms require input from users. If you need information that can be represented as a choice of multiple values, use input with type 'radio'.
+ * Help prop provides guidance to ensure they know what to enter.
+ *
+ * Radios need to be wrapped in RadioGroup component that provides radio buttons with context so that they behave as one component.
+ */
+export const Radio: FC<RadioProps> = ({
+  onBlur,
+  className,
+  value,
+  label,
+  required,
+  helperText,
+  disabled = false,
+  checked,
+  'data-test': dataTest,
+}) => {
+  const idRef = useRef(uuid())
+
+  return (
+    <RadioGroupContext.Consumer>
```

Comment:
nit: Since we are using hooks everywhere I would suggest utilizing `useContext` instead